### ISSUE TITLE
Math Parser Idea: e(value,n)

### ISSUE THREAD

**Issue by Reptorian1125** [GreycLab/CImg#433]
There are sometimes when I had to use a workaround to the fact that you cannot place variables next to e. 2.4e$var isn't allowed.

```
foo:
m=3
eval "2.5e$m;"
```
This will give a error.

So, one way of solving it is e(value,n). Which is the equivalent of {value}e{n} where value, and n is a number. Or maybe unnecessary, and there should be some fixes to the parsing of 2.5e$m. 

**EDIT:** {value}*10^{n} will do too, but ideally I would like to see better parsing.

**Comment by dtschump** [comment]
Variable are not substituted when inside double-quotes, so you have to move them outside double quotes, like this:
~~~
foo:
  m=3
  eval "2.5e"$m";"
~~~

**Comment by Reptorian1125** [comment]
Yes, I know.

 I thought it was possible with 0b, and 0x. I checked, and it turns out they only work with $1,$-1. 

In Lavender Binary Map case:
```
		const first_binary_to_search=0b$3;
		const second_binary_to_search=0b$4;
```
That does work.

**Comment by dtschump** [comment]
Yes, that's because `$1` is actually not a variable, it's a command argument, which is quite different (even if the syntax is similar).
- A command argument as `$1` is indeed substituted inside double-quoted strings.
- A variable as `$foo` is not.

**Comment by Reptorian1125** [comment]
Make a new issue to discuss that. That is offtopic.
